src/Workspaces/Core/Portable/Shared/Extensions/SyntaxEditorExtensions.cs

```diff
@@ -29,8 +52,32 @@ internal static class SyntaxEditorExtensions
                 editor, document,
                 originalNodes,
                 (syntaxFacts, node) => GetExpressionSemanticBoundary(syntaxFacts, node),
+                (semanticModel, _, node) => canReplace(semanticModel, (TNode)node),
+                (semanticModel, currentRoot, _, node) => updateRoot(semanticModel, currentRoot, (TNode)node),
+                (t) => t,
+                cancellationToken);
```

Comment:
the direction here seems reversed.  i would expect this to call into the new, more general, ApplyExpressionLevelSemanticEditsAsync.  And that new function is the one that will call into ApplySemanticEditsAsync.

Basically, more specific functions keep calling the more general ones.  so it would go OldApplyExpressionLevelSemanticEditsAsync -> NewApplyExpressionLevelSemanticEditsAsync -> ApplySemanticEditsAsync

Reply:
Like